labs/RivePlayer/src/RivePlayer.Platform.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace CommunityToolkit.Labs.WinUI.Rive;
+
+// This file contains platform-specific customizations of RivePlayer.
+
+#if WINDOWS_WINAPPSDK
+
+using SkiaSharp.Views.Windows;
+
+// WinAppSdk doesn't have SKSwapChainPanel yet.
+public partial class RivePlayer : SKXamlCanvas, IDisposable
+{
+    // SKXamlCanvas doesn't support rendering in a background thread.
+    public bool DrawInBackground { get; set; }
+
+    private void OnPaintSurface(object? sender, SKPaintSurfaceEventArgs e)
+    {
+        this.PaintNextAnimationFrame(e.Surface, e.Info.Width, e.Info.Height);
+    }
+#else
+
+// SkiaSharp.Views.UWP is on Uno too.
+using SkiaSharp.Views.UWP;
+
+// SKSwapChainPanel performs better than SKXamlCanvas.
+public partial class RivePlayer : SKSwapChainPanel, IDisposable
+{
+    private void OnPaintSurface(object? sender, SKPaintGLSurfaceEventArgs e)
+    {
+        this.PaintNextAnimationFrame(e.Surface, e.BackendRenderTarget.Width, e.BackendRenderTarget.Height);
+    }
+#endif
+
+#if HAS_UNO
+    // FrameworkElement has a method named Dispose in Uno, which prevents us from implementing
+    // IDisposable.Dispose. We just have to wait until the destructor runs.
```

Comment:
Any C# experts know how to properly resolve this? I've tried:

```
void IDisposable.Dispose()
  > Rename 'RivePlayer.System.IDisposable.Dispose' to 'Dispose' and ensure that it is declared as public and sealed

public sealed void Dispose()
  > 'RivePlayer.Dispose()' hides inherited member 'FrameworkElement.Dispose()'. Use the new keyword if hiding was intended.

public sealed new void Dispose()
  > 'RivePlayer.Dispose()' cannot be sealed because it is not an override 
  > The member 'RivePlayer.Dispose()' does not hide an accessible member. The new keyword is not required.
  > The member 'RivePlayer.Dispose()' does not hide an accessible member. The new keyword is not required.

```

This works:

```
public new void Dispose()
```

But I suspect with the `new` keywork it actually isn't overriding `IDisposable.Dispose`.

Maybe Uno's `FrameworkElement` sends out a notification we could register to?

We get a warning if RivePlayer doesn't inherit from IDisposable because it owns disposable members.

Reply:
@jeromelaban is this something you can help clarify for us?

Reply:
Types coming from Uno are implementing `Dispose` as a result of .NET iOS and Android bindings, but actually invoking `Dispose` is causing a large number issues with the lifecycle of underlying native instances. 

This is why Uno is explicitly blocking inherited types from overriding `Dispose` as a best effort, and it's also important to note that corresponding WinUI controls do not provide Dispose either, so the code invoking `Dispose` would not work when compiled for Windows (WinUI or UW).

Reply:
Thanks @jeromelaban any recommendations for handling this, I know the lifecycle events of loading/unloading are fickle, so trying to unload/re-initialize based on that instead may be tricky.

(Also, we were having issues with `SKSwapChainPanel` in WASM, should that be working there as well? `SKXamlCanvas` seemed fine.)

Reply:
Load/Unload is the only way to handle this, unfortunately, in the same way handling native dependencies work with UWP/WinAppSDK. Otherwise you can try relying on the finalizer, but in this case, you may be having to deal memory leaks and thread affinity to the dispatcher.

For SKSwapChainPanel, this is supposed to work properly (an example [here](https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest)).

Reply:
Ah @csmartdalton looks like there's a 2nd nuget package we need for webassembly:

```xml
    <PackageReference Include="SkiaSharp.NativeAssets.WebAssembly" Version="2.88.0-preview.232" />
```

@Arlodotexe do we have a Labs flag we can use for the dependencies for a conditional here to include a package just for Uno WASM?

Reply:
I tried adding SkiaSharp.NativeAssets.WebAssembly and it looked like it was already bubbling in through one of the SkiaSharp dependencies.

Also, I just got rid of disposable objects for now. We can start adding them back after this has landed and had a chance to start soaking.